src/norm/pragmas.nim

```diff
@@ -19,3 +19,80 @@ template ro* {.pragma.}
 
 template unique* {.pragma.}
   ## Mark field as unique.
+
+template fk*(val: typed) {.pragma.}
+  ##[ Mark ``int`` field as foreign key. Foreign keys always references the field ``id`` of ``val``. ``val`` should be a Model.
+  ]##
+runnableExamples:
+  import os
+  import norm/model
+  import norm/sqlite
+  import norm/pragmas
+
+  type
+    Product = ref object of Model
+      name : string
+      price : float
+
+    Consumer = ref object of Model
+      name: string
+      productId {.fk: Product.}: int
+
+  proc newProduct(): Product=
+    Product(name: "", price: 0.0)
+
+  proc newConsumer(name: string = "", productId: int = 0): Consumer=
+    Consumer(name: name, productId: productId)
+
+  let dbName = getTempDir() / "example.db"
+  # Clean db
+  discard tryRemoveFile(dbName)
+  let db = open(dbName, "", "", "")
+  # Depending on how sqlite3 was compiled, enabling foreign_keys may be necessary
+  db.exec(sql"PRAGMA foreign_keys = ON")
+
+  block:
```

Comment:
I see you use those block based tests quite often, which is not in line with the rest of the tests. I'm not against that style per se but I'd rather have all tests sharing one consistent structure.